src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -3,16 +3,19 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
-#if !NETSTANDARD2_0
+using System.Runtime.CompilerServices;
+
 using Internal.Runtime.CompilerServices;
-#endif
 
 namespace System
 {
     internal static partial class SpanHelpers // .T
     {
         public static int IndexOf<T>(ref T searchSpace, int searchSpaceLength, ref T value, int valueLength) where T : IEquatable<T>
         {
+            // The optimized implementation should be used for these types
+            Debug.Assert(!RuntimeHelpers.IsBitwiseEquatable<T>() || !(Unsafe.SizeOf<T>() == sizeof(byte) || Unsafe.SizeOf<T>() == sizeof(char)));
```

Comment:
Interested about this: what can cause non-bitwise equatable when the size is less than pointer?

Reply:
For example:

```
struct MyOneByteStruct : IEquatable<T>
{ 
    byte _flags;

    bool Equals(T other)
    {
        // The structs are considered equal when second bit of flags is equal
        return (_flags & 2) == (other._flags & 2);
    }
}
```